validators/email.py

```diff
@@ -1,75 +1,61 @@
-    # quoted-string
-    r'|^"([\001-\010\013\014\016-\037!#-\[\]-\177]|'
-    r"""\\[\001-\011\013\014\016-\177])*"$)""",
-    re.IGNORECASE
-)
-domain_regex = re.compile(
-    # domain
-    r'(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+'
-    r'(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?$)'
-    # literal form, ipv4 address (SMTP 4.1.3)
-    r'|^\[(25[0-5]|2[0-4]\d|[0-1]?\d?\d)'
-    r'(\.(25[0-5]|2[0-4]\d|[0-1]?\d?\d)){3}\]$',
-    re.IGNORECASE)
-domain_whitelist = ['localhost']
+from .domain import domain
 
 
 @validator
-def email(value, whitelist=None):
-    """
-    Validate an email address.
+def email(value: str, /):
+    """Validate an email address.
 
-    This validator is based on `Django's email validator`_. Returns
-    ``True`` on success and :class:`~validators.utils.ValidationFailure`
-    when validation fails.
+    This was inspired from [Django's email validator][1].
+    Also ref: [RFC 1034][2], [RFC 5321][3] and [RFC 5322][4].
 
-    Examples::
+    [1]: https://github.com/django/django/blob/main/django/core/validators.py#L174
+    [2]: https://www.rfc-editor.org/rfc/rfc1034
+    [3]: https://www.rfc-editor.org/rfc/rfc5321
+    [4]: https://www.rfc-editor.org/rfc/rfc5322
 
+    Examples:
         >>> email('someone@example.com')
-        True
-
+        # Output: True
         >>> email('bogus@@')
-        ValidationFailure(func=email, ...)
-
-    .. _Django's email validator:
-       https://github.com/django/django/blob/master/django/core/validators.py
+        # Output: ValidationFailure(email=email, args={'value': 'bogus@@'})
 
-    .. versionadded:: 0.1
+    Args:
+        value:
+            eMail string to validate.
 
-    :param value: value to validate
-    :param whitelist: domain names to whitelist
+    Returns:
+        (Literal[True]):
+            If `value` is a valid eMail.
+        (ValidationFailure):
+            If `value` is an invalid eMail.
 
-    :copyright: (c) Django Software Foundation and individual contributors.
-    :license: BSD
+    > *New in version 0.1.0*.
     """
-
-    if whitelist is None:
-        whitelist = domain_whitelist
-
-    if not value or '@' not in value:
+    if not value or value.count("@") != 1:
         return False
 
-    user_part, domain_part = value.rsplit('@', 1)
-
-    if not user_regex.match(user_part):
-        return False
+    username_part, domain_part = value.rsplit("@", 1)
 
-    if len(user_part.encode("utf-8")) > 64:
+    if len(username_part) > 64 or len(domain_part) > 253:
+        # ref: RFC 1034 and 5231
         return False
 
-    if domain_part not in whitelist and not domain_regex.match(domain_part):
-        # Try for possible IDN domain-part
-        try:
-            domain_part = domain_part.encode('idna').decode('ascii')
-            return domain_regex.match(domain_part)
-        except UnicodeError:
-            return False
-    return True
+    return (
+        bool(domain(domain_part))
+        if re.compile(
```

Comment:
Wouldn't it be better to have the user_regex defined in the variable as it was previously? This way it doesn't need to be compiled everytime this function is called. Also some users of this library might want to use re-use this regex for other purposes.

Reply:
Please ref: https://github.com/python-validators/validators/pull/232#discussion_r1124219623